app/controllers/reports/match_progress_controller.rb

```diff
@@ -0,0 +1,119 @@
+###
+# Copyright 2016 - 2020 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module Reports
+  class MatchProgressController < ApplicationController
+    include ArelHelper
+
+    before_action :authenticate_user!
+
+    def index
+      respond_to do |format|
+        format.html {}
+        format.xlsx do
+          @included_sub_programs = sub_program_list.invert.slice(*report_params[:sub_programs])
+          filename = 'CAS Match Progress.xlsx'
+          render xlsx: 'index.xlsx', filename: filename
+        end
+      end
+    end
+
+    def actions(sub_program_id)
+      # {match_id => {decision_order => [[date, text], ...]}}
+      matches = ClientOpportunityMatch.
+        joins(:opportunity).
```

Comment:
why not `joins(opportunity: :voucher).merge(Voucher.where(sub_program_id: sub_program_id)))`?